rafthttp/snapshot_hub.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rafthttp
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/coreos/etcd/Godeps/_workspace/src/golang.org/x/net/context"
+	"github.com/coreos/etcd/pkg/types"
+	"github.com/coreos/etcd/raft"
+	"github.com/coreos/etcd/raft/raftpb"
+)
+
+// timeout that waits snapshot to be called to write out
+// Before snapshot is called to write out, raft snapshot message needs to be sent out,
+// then let remote member send request to ask for snapshot, so at least
+// two maximal RTT is required, which is 10s.
+const snapshotWaitTimeout = 10 * time.Second
+
+type snapshotHub struct {
+	tr         http.RoundTripper
+	urlPickers map[types.ID]*urlPicker
+	r          Raft
+	snapst     SnapshotStore
+
+	// protect fields below
+	mu sync.Mutex
+	// whether it has registered snapshot that is going to be written out or closed
```

Comment:
Why `that is going to be written out or closed` is important to this variable? I cannot see a clear relationship between this boolean variable and the comment.